src/main/java/com/powsybl/openloadflow/dc/DcLoadFlowEngine.java

```diff
@@ -71,45 +75,73 @@ public DcLoadFlowResult run() {
         // only process main (largest) connected component
         LfNetwork network = networks.get(0);
 
-        if (parameters.isDistributedSlack()) {
-            distributeSlack(network);
-        }
-
         DcEquationSystemCreationParameters creationParameters = new DcEquationSystemCreationParameters(parameters.isUpdateFlows(), false, parameters.isForcePhaseControlOffAndAddAngle1Var(), parameters.isUseTransformerRatio());
         EquationSystem equationSystem = DcEquationSystem.create(network, new VariableSet(), creationParameters);
 
+        LoadFlowResult.ComponentResult.Status status = LoadFlowResult.ComponentResult.Status.FAILED;
+        try (JacobianMatrix j = new JacobianMatrix(equationSystem, parameters.getMatrixFactory())) {
+
+            status = run(equationSystem, j, Collections.emptyList());
+            stopwatch.stop();
+            LOGGER.debug(Markers.PERFORMANCE_MARKER, "Dc loadflow ran in {} ms", stopwatch.elapsed(TimeUnit.MILLISECONDS));
+
+        } catch (Exception e) {
+            LOGGER.error("Failed to solve linear system for DC load flow", e);
+        }
+
+        return new DcLoadFlowResult(network, getActivePowerMismatch(network.getBuses()), status);
+    }
+
+    public LoadFlowResult.ComponentResult.Status run(EquationSystem equationSystem, JacobianMatrix j, Collection<LfBus> removedBuses) {
+
         double[] x = equationSystem.createStateVector(new UniformValueVoltageInitializer());
 
-        equationSystem.updateEquations(x);
+        // only process main (largest) connected component
+        LfNetwork network = networks.get(0);
 
-        double[] targets = equationSystem.createTargetVector();
+        Collection<LfBus> remainingBuses = new HashSet<>(network.getBuses());
+        remainingBuses.removeAll(removedBuses);
 
-        try (JacobianMatrix j = new JacobianMatrix(equationSystem, parameters.getMatrixFactory())) {
-            double[] dx = Arrays.copyOf(targets, targets.length);
+        if (parameters.isDistributedSlack()) {
+            distributeSlack(remainingBuses);
+        }
 
-            LoadFlowResult.ComponentResult.Status status;
-            try {
-                j.solveTransposed(dx);
-                status = LoadFlowResult.ComponentResult.Status.CONVERGED;
-            } catch (Exception e) {
-                status = LoadFlowResult.ComponentResult.Status.FAILED;
-                LOGGER.error("Failed to solve linear system for DC load flow", e);
-            }
+        equationSystem.updateEquations(x);
 
-            equationSystem.updateEquations(dx);
-            equationSystem.updateNetwork(dx);
+        this.targetVector = equationSystem.createTargetVector();
 
-            // set all calculated voltages to NaN
-            for (LfBus bus : network.getBuses()) {
-                bus.setV(Double.NaN);
-            }
+        if (!removedBuses.isEmpty()) {
+            // set buses injections and transformers to 0
```

Comment:
Losing a transformer in a contingency requires us to do
equationSystem.getEquation(transformer.getNum(), EquationType.BRANCH_ALPHA1).get().getColumn() and set this column to 0.
So we may also need to pass a list of removedTransformers

Reply:
(I'll do it in another PR and ask for a merge on this one)

Reply:
(see #237 )